java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -2327,24 +2327,102 @@ public final ColumnVector stringConcatenateListElements(Scalar separator,
+   *
+   * The caller is responsible for checking the output column size will not exceed the maximum size
+   * of a strings column (number of total characters is less than the value {@link Integer#MAX_VALUE}).
+   *
+   * @param repeatTimes The column containing numbers of times each input string is repeated.
+   * @return A new java column vector containing repeated strings.
+   */
+  public final ColumnVector repeatStrings(ColumnView repeatTimes) {
+    assert type.equals(DType.STRING) : "column type must be String";
+    return new ColumnVector(repeatStringsWithColumnRepeatTimes(getNativeView(),
+            repeatTimes.getNativeView(), 0));
+  }
+
+  /**
+   * This function is an overloaded version of {@link #repeatStrings(ColumnView) repeatStrings},
+   * with an additional parameter <code>outputStringSizes</code> that provides a column containing
+   * the pre-computed sizes of the output strings.
+   *
+   * @param repeatTimes The column containing numbers of times each input string is repeated.
+   * @param outputStringSizes A numeric column providing the pre-computed sizes of the output strings.
+   * @return A new java column vector containing repeated strings.
+   */
+  public final ColumnVector repeatStrings(ColumnView repeatTimes, ColumnView outputStringSizes) {
+    assert type.equals(DType.STRING) : "column type must be String";
+    return new ColumnVector(repeatStringsWithColumnRepeatTimes(getNativeView(),
+            repeatTimes.getNativeView(), outputStringSizes.getNativeView()));
+  }
+
+  /**
+   * Struct to hold the results output from the function
+   * {@link #computeOutputSizesRepeatStrings(ColumnView) computeOutputSizesRepeatStrings}.
+   */
+  static final class OutputSizesRepeatStrings implements AutoCloseable {
+    public final ColumnVector stringSizes;
+    public final long totalStringSize;
+    OutputSizesRepeatStrings(ColumnVector stringSizes, long totalStringSize) {
+      this.stringSizes = stringSizes;
+      this.totalStringSize = totalStringSize;
+    }
+
+    /** Close the underlying resources */
+    @Override
+    public void close() {
+      if (stringSizes != null) {
+        stringSizes.close();
+      }
+    }
+  };
+
+  /**
+   * Compute sizes of the output strings if each string in an input strings column is repeated by
+   * a different number of times given by the corresponding row in a <code>repeatTimes</code>
+   * numeric column (i.e., compute sizes of the strings resulted from
+   * {@link #repeatStringsWithColumnRepeatTimes}).
+   *
+   * @param repeatTimes The column containing numbers of times each input string is repeated.
+   * @return An instance of OutputSizesRepeatStrings class which stores a Java column vector containing
+   *         the computed sizes of the repeated strings, and a long value storing sum of all these
+   *         computed sizes.
+   */
+  public final OutputSizesRepeatStrings computeOutputSizesRepeatStrings(ColumnView repeatTimes) {
```

Comment:
Nit: This is quite wordy, how about something shorter like `repeatStringsWithSizes`?

Reply:
The output is just sizes, not any string repeating. So we don't use "With". Maybe `outputSizesRepeatedStrings` which is shorter?

Reply:
Ah, sorry, forgot it doesn't have any repeats in it.

I think the "output" part is redundant.  Any returned result is an output, but we don't put `output` on every method name.  `repeatStringsSizes` is nice and short and I think conveys the idea.  Naming is hard. 😅